pkg/asset/machines/gcp/machines.go

```diff
@@ -77,6 +77,22 @@ func provider(clusterID string, platform *gcp.Platform, mpool *gcp.MachinePool,
 		return nil, err
 	}
 
+	var encryptionKey *gcpprovider.GCPEncryptionKeyReference
+
+	if mpool.OSDisk.EncryptionKey != nil {
+		encryptionKey = &gcpprovider.GCPEncryptionKeyReference{
+			KMSKey: &gcpprovider.GCPKMSKeyReference{
+				Name:      mpool.OSDisk.EncryptionKey.KMSKey.Name,
+				KeyRing:   mpool.OSDisk.EncryptionKey.KMSKey.KeyRing,
+				ProjectID: mpool.OSDisk.EncryptionKey.KMSKey.ProjectID,
+				Location:  mpool.OSDisk.EncryptionKey.KMSKey.Location,
+			},
+			KMSKeyServiceAccount: mpool.OSDisk.EncryptionKey.KMSKeyServiceAccount,
+		}
+	} else {
+		encryptionKey = nil
```

Comment:
nit: not sure this is necessary (should already be nil). if you want to leave it in, moving it out of `else` and before `if` would be more concise.

Reply:
This shouldn't be needed as `var encryptionKey *gcpprovider.GCPEncryptionKeyReference` above the if makes it a nil anyway